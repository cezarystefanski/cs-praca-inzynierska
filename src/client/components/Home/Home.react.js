import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ROUTE_NAMES } from '../../constants';
import generateTitle from '../../helpers';

import classes from './Home.scss';

const componentTitle = 'Home';

class Home extends Component {
  constructor(props) {
    super();
    this.props = props;
  }

  componentDidMount() {
    this.props.onHomeVisit();
  }

  render() {
    const init = 'Initial stuff';
    const {
      homeVisits,
      onResetClick,
    } = this.props;

    return (
      <div>
        {generateTitle(componentTitle)}
        <h1 className={classes.heading}>React App starting point</h1>
        <h2>{init}</h2>
        <small>Times homepage was visited: <span className="numHomeVisits">{homeVisits}</span></small>
        <button onClick={onResetClick}>Reset home visits number</button>
        <p>
          <Link to={ROUTE_NAMES.example}>Link to examples</Link>
        </p>
      </div>
    );
  }
}

Home.propTypes = {
  homeVisits: PropTypes.number.isRequired,
  onHomeVisit: PropTypes.func.isRequired,
  onResetClick: PropTypes.func.isRequired,
};

export default Home;
