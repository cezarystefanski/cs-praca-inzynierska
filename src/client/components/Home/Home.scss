.heading {
  color: rgb(177, 204, 82);
}
