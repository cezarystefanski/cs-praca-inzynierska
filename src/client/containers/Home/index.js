import { connect } from 'react-redux';
import { Home } from '../../components';
import {
  resetHomeVisits,
  increaseHomeVisits,
} from '../../actions';

const mapStateToProps = state => ({
  homeVisits: state.homeVisits,
});

const mapDispatchToProps = dispatch => ({
  onHomeVisit() {
    return dispatch(increaseHomeVisits());
  },
  onResetClick() {
    return dispatch(resetHomeVisits());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Home);
