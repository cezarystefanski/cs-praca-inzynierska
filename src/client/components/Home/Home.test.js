import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Home from './Home.react';

const homeMock = renderer
  .create((
    <MemoryRouter>
      <Home homeVisits={0} onResetClick={() => {}} onHomeVisit={() => {}} />
    </MemoryRouter>
  ))
  .toJSON();

describe('home component', () => {
  it('renders correctly', () => {
    expect(homeMock).toMatchSnapshot();
  });
});
