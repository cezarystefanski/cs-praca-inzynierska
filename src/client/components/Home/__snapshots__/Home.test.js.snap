// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component renders correctly 1`] = `
<div>
  <h1
    className="heading"
  >
    React App starting point
  </h1>
  <h2>
    Initial stuff
  </h2>
  <small>
    Times homepage was visited: 
    <span
      className="numHomeVisits"
    >
      0
    </span>
  </small>
  <button
    onClick={[Function]}
  >
    Reset home visits number
  </button>
  <p>
    <a
      href="/example"
      onClick={[Function]}
    >
      Link to examples
    </a>
  </p>
</div>
`;
